src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/MigrateHandlerInterceptorTest.java

```diff
@@ -114,6 +134,25 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
                   }
               }
               """
+          ),
+          // And do not change any other super call when HandlerInterceptorAdapter is imported
+          java(
+            """
+              import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+              class B extends A {
+                  @Override
+                  public String test() {
+                      return super.test();
+                  }
+              }
+
+              class A {
+                  public String test() {
+                      return "test";
+                  }
+              }
+              """
```

Comment:
Just because there's an import does not mean the type is in use on any LST elements. As such the precondition will fail here, and hence no changes will be applied, whereas you probably intended to test that no changes are made even when the precondition matches; can you either remove this block or ensure the precondition matches and no change is made?

Reply:
Ah good one, changed it!